Code.gs

```diff
@@ -178,46 +171,82 @@ function main(){
     }
   }
   //----------------------------------------------------------------
-
-
-
-  //-------------- Remove Or modify events from calendar -----------  
+  
+  
+  //-------------- Remove Or modify events from calendar -----------
+  var alreadyProcessedFids = [];
   Logger.log("Checking " + calendarEvents.length + " events for removal or modification");
   for (var i = 0; i < calendarEvents.length; i++){
     var tagValue = calendarEvents[i].getTag("FID");
     var feedIndex = feedEventIds.indexOf(tagValue);
     
+    if (alreadyProcessedFids.indexOf(tagValue) > -1)
+      continue;
+    
     if(removeEventsFromCalendar){
       if(feedIndex  == -1 && tagValue != null){
-        Logger.log("    Deleting " + calendarEvents[i].getTitle());
-        calendarEvents[i].deleteEvent();
+        Logger.log("    Deleting " + calendarEvents[i].getTitle() + " (id: " + tagValue + ")");
+        calendarEvents[i].getEventSeries().deleteEventSeries(); //Delete the event by series. This works even if it is not a recurring event 
       }
     }
 
     if(modifyExistingEvents){
       if(feedIndex != -1){
-        var e = calendarEvents[i];
-        var fes = events.filter(sameEvent, calendarEvents[i].getTag("FID"));
+        Logger.log("    Checking for modification " + calendarEvents[i].getTitle() + " (id: " + tagValue + ")");
         
-        if(fes.length > 0){
-          var fe = fes[0];
-
+        var fes = icsEvents.filter(sameEvent, calendarEvents[i].getTag("FID"));
+        
+        if (fes.length <= 0)
+          continue;
+          
+        var fe = fes[0];
+        
+        if (fe.recurrence != null){
+          var eSeries = calendarEvents[i].getEventSeries();
+          
+          //Since there is no CalendarEventSeries.getRecurrence() method we can use to check then we will always set the recurrence
+          //(which also means setting the time). This also solves the situation of a regular event becoming a recurring event.
+          eSeries.setRecurrence(fe.recurrence, fe.startTime, fe.endTime);
+          
+          if (eSeries.getTitle() != fe.title)
+            eSeries.setTitle(fe.title);
+          
+          if (eSeries.getLocation() != fe.location)
+            eSeries.setLocation(fe.location);
+            
+          if (eSeries.getDescription() != fe.description)
+            eSeries.setDescription(fe.description);
+        }
+        else{
+          var e = calendarEvents[i];
+          
+          var occurences = calendarFids.filter(function(value){ return value === tagValue; }).length;
+          if (occurences > 1){
+            //Event used to be a recurring event, but is now a regular event. Delete the recurring series from
+            //the calendar and create the new event fresh
+            e.getEventSeries().deleteEventSeries();
+            CreateEvent(targetCalendar, fe);
```

Comment:
Have to set the ID of the new event

Reply:
The checks below should be the else case, no need to check for changes after freshly adding the new event